### PR TITLE
Strip spaces and any params from media type when generating an RDF type

### PR DESCRIPTION
**Pull request by timbl**
Strip spaces and any params from media type when generating an RDF type
Add content type default of turtle to rdf.parse

### CODE REVIEW

File: src/named-node.js [hunk 1]
```diff
@@ -15,7 +15,13 @@ class NamedNode extends Node {
     super()
     this.termType = NamedNode.termType
     if (iri.indexOf(':') < 0) {
-      console.log('Warning: NamedNode IRI "' + iri + '" must be absolute. Relative URIs will fail in future versions')
+      console.log('Error: NamedNode IRI "' + iri + '" must be absolute. Relative URIs will fail in future versions')
+    }
+    if (iri.indexOf(' ') >= 0) {
```
Comment (dmitrizagidulin) [inline src/named-node.js]: maybe `iri.includes(' ')` might be slightly clearer/more idiomatic?

**Comment by dmitrizagidulin** [comment]
👍 , looks fine to me.

File: src/named-node.js [hunk 2]
```diff
@@ -15,7 +15,13 @@ class NamedNode extends Node {
...
+      var message = 'Error: NamedNode IRI "' + iri + '" must not contain unencoded spaces.'
+      console.log(message)
+      // throw new Error(message)
+      iri = encodeURI(iri)
```
Comment (dmitrizagidulin) [inline src/named-node.js]: Er, wait a second, should this be the right behavior? To automatically `encodeURI` the full node uri?

What's the use case you're encountering, when spaces are being passed in to a named node?

File: src/fetcher.js [hunk 3]
```diff
@@ -586,6 +586,7 @@ var Fetcher = function Fetcher (store, timeout, async) {
   //  Writes back to the web what we have in the store for this uri
   this.putBack = function (uri, options) {
     uri = uri.uri || uri // Accept object or string
+    options = options || {}
```
Comment (dan-f) [inline src/fetcher.js]: Can alternatively use [ES2015 default argument syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

File: src/named-node.js [hunk 4]
```diff
@@ -15,7 +15,13 @@ class NamedNode extends Node {
     super()
     this.termType = NamedNode.termType
     if (iri.indexOf(':') < 0) {
-      console.log('Warning: NamedNode IRI "' + iri + '" must be absolute. Relative URIs will fail in future versions')
+      console.log('Error: NamedNode IRI "' + iri + '" must be absolute. Relative URIs will fail in future versions')
+    }
+    if (iri.indexOf(' ') >= 0) {
+      var message = 'Error: NamedNode IRI "' + iri + '" must not contain unencoded spaces.'
```
Comment (dan-f) [inline src/named-node.js]: It could be worth it to use a URL validation library rather these ad-hoc checks.

File: src/named-node.js [hunk 5]
```diff
@@ -15,7 +15,13 @@ class NamedNode extends Node {
...
+      console.log(message)
```
Comment (dan-f) [inline src/named-node.js]: Why `console.log()` the message? It seems like constructing a named node from an invalid URI should simply be an error.

File: src/named-node.js (hunk 4 above)
Comment (dmitrizagidulin) [inline src/named-node.js]: Agreed. Shall we open an issue for that?

File: src/fetcher.js (hunk 3 above)
Comment (dmitrizagidulin) [inline src/fetcher.js]: Ok, changed it to use default argument syntax.

**Comment by dmitrizagidulin** [comment]
Added tests.

File: src/named-node.js (hunk 4 above)
Comment (dan-f) [inline src/named-node.js]: I mean, might as well do it now? I don't particularly mind either way.

Comment (dmitrizagidulin) [inline src/named-node.js]: I opened issue #168 as a reminder.

**Pull request merged**